frontend/src/store/modules/eceweApp.js

```diff
@@ -1,153 +1,137 @@
+          return;
+        } else {
+          console.info('ECEWEApp.saveECEWE: model changes (SAVING).');
         }
-        let payload = JSON.parse(JSON.stringify(state));
-        
-        // remove attributes we are not updating before sending payload.
+        let payload = { ...state.eceweModel };
+        payload = JSON.parse(JSON.stringify(payload));
         delete payload.facilities;
         // Save ECEWE parent record.
-        let response = await ApiService.apiAxios.patch(ApiRoutes.APPLICATION_ECEWE + '/' + rootState.organization.applicationId, payload);
-
-        // If parent ECEWE save was successfull and ECEWE facilties have been flagged for update, issue update on ECEWE facilties as their
-        // previous optinout responses are no longer applicable.
-        if (response?.status == '200' && updateFacilities) {
-          dispatch('saveECEWEFacilities');
-        }
+        let response = await ApiService.apiAxios.patch(ApiRoutes.APPLICATION_ECEWE + '/' + state.applicationId, payload);
+        commit('setLoadedModel', {...state.eceweModel});
+        commit('setEceweEligibilityComplete', payload.optInECEWE != null);
         return response;
       } catch (error) {
-        console.log(`Failed to update existing ECEWE application - ${error}`);
+        console.info(`Failed to update existing ECEWE application - ${error}`);
         throw error;
       }
     },
-    async saveECEWEFacilities({ state, rootState, commit }) {
-      if (!localStorage.getItem('jwtToken')) { // DONT Call api if there is no token.
-        console.log('unable to save because you are not logged in');
-        throw new Error('unable to save ecewe facility application because you are not logged in');
-      }
-      let facilitiesForBackend = state.facilities;
-      let payload = JSON.parse(JSON.stringify(facilitiesForBackend));
+    async saveECEWEFacilities({ state, commit }) {
+      checkSession();
+      console.info('ECEWEApp.saveECEWEFacilities.state.loadedFacilities = ');
+      console.info(JSON.parse(JSON.stringify(state.loadedFacilities)));
+      console.info('ECEWEApp.saveECEWEFacilities.state.facilities = ');
+      console.info(JSON.parse(JSON.stringify(state.facilities)));
+      if (isEqual(state.loadedFacilities, state.facilities)) {
+        console.info('ECEWEApp.saveECEWEFacilities: no model changes (NOT saving).');
+        return;
+      } else {
+        console.info('ECEWEApp.saveECEWEFacilities: model changes (SAVING).');
+      }      
+      let payload = JSON.parse(JSON.stringify(state.facilities));
       try {
-        let response = await ApiService.apiAxios.post(ApiRoutes.APPLICATION_ECEWE_FACILITY + '/' + rootState.organization.applicationId, payload);
+        let response = await ApiService.apiAxios.post(ApiRoutes.APPLICATION_ECEWE_FACILITY + '/' + state.applicationId, payload);
         commit('setFacilities', response.data.facilities);
+        commit('setLoadedFacilities', response.data.facilities);
```

Comment:
can loaded facilities be set before the post, and can we just have it be a copy of state.facilities?